cpp/tests/utilities/column_utilities.cu

```diff
@@ -46,78 +47,310 @@
 #include "rmm/cuda_stream_view.hpp"
 
 namespace cudf {
+
 namespace test {
 
 namespace {
 
+// expand all non-null rows in a list column into a column of child row indices.
+std::unique_ptr<column> generate_child_row_indices(lists_column_view const& c,
+                                                   column_view const& row_indices)
```

Comment:
Do we need an assertion for `row_indices` not having nulls? Overkill for `cudf::test` code?

Reply:
P.S. This was an informative read.

Reply:
I think it's overkill here.  `row_indices` is never something handed to us by the external user.  It's purely internal.